### DISCUSSION TITLE
seems like despawn also despawn children?

### DISCUSSION THREAD

**Discussion by alexniver** [bevyengine/bevy#10867] [Q&A] [answered]
I'm not sure is this a bug or I'm doing something wrong
```
use bevy::{prelude::*, window::close_on_esc};
use bevy_inspector_egui::quick::WorldInspectorPlugin;

fn main() {
    App::new()
        .add_plugins((DefaultPlugins, WorldInspectorPlugin::new()))
        .add_systems(Update, (close_on_esc, spawn_p_c, despawn_p_only))
        .run();
}

#[derive(Debug, Component)]
struct P;

#[derive(Debug, Component)]
struct C;

fn spawn_p_c(mut commands: Commands, input: ResMut<Input<KeyCode>>) {
    if input.just_pressed(KeyCode::S) {
        commands.spawn(P).with_children(|c| {
            c.spawn(C);
        });
    }
}

fn despawn_p_only(
    mut commands: Commands,
    query: Query<Entity, With<P>>,
    input: ResMut<Input<KeyCode>>,
) {
    if input.just_pressed(KeyCode::D) {
        for p_entity in &query {
            commands.entity(p_entity).despawn();
        }
    }
}
```

I want just despawn `P`, but it's children 'C' also despawn.
![Peek 2023-12-05 01-48](https://github.com/bevyengine/bevy/assets/3366229/28408eb1-7cf6-4b70-b69e-09a8f07d7d1f)

bevy version : 0.12.1

**Comment by rparrett** [comment]
I think this is more likely to be an issue with `bevy_inspector_egui`. Could you try removing that plugin from your example and seeing if the behavior is the same?

**Comment by SkiFire13** [comment, marked as answer]
When you despawn the entities with the `P` component you don't fixup the `Parent` component of their children, which remains there pointing now invalid entities. Then `bevy_inspector_egui` shows only entities without a `Parent` component, which filters out your entities with a `C` component.

Consider calling [`.clear_children()`](https://docs.rs/bevy_hierarchy/latest/bevy_hierarchy/trait.BuildChildren.html#tymethod.clear_children) before `.despawn()`:

```diff
        for p_entity in &query {
-            commands.entity(p_entity).despawn();
+            commands.entity(p_entity).clear_children().despawn();
        }
```

**Reply by alexniver** [reply]
Yeah! you are right!
